staging/src/k8s.io/client-go/tools/events/interfaces.go

```diff
@@ -68,3 +69,15 @@ type EventSink interface {
 	Update(event *v1beta1.Event) (*v1beta1.Event, error)
 	Patch(oldEvent *v1beta1.Event, data []byte) (*v1beta1.Event, error)
 }
+
+// EventClient knows whether to use new or old Event API for handling events.
+type EventClient interface {
```

Comment:
EventClient name is misleading, because it's not a client per se.

Maybe EventBroadcasterAdapter ?

```
// EventBroadcasterAdapter is a auxiliary interface to simplify migration to
// the new events API. It is a wrapper around new and legacy broadcasters
// that smartly chooses which one to use.
//
// Deprecated: This interface will be removed once migration is completed.
type EventBroadcasterAdapter interface {
```

Reply:
Done.